common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -318,7 +318,7 @@ public String toString() {
 
   private void appendUnit(StringBuilder sb, long value, String unit) {
     if (value != 0) {
-      sb.append(" " + value + " " + unit + "s");
+      sb.append(" ").append(value).append(" ").append(unit).append("s");
```

Comment:
Super nit, but if you're optimizing this, append characters instead of single-character strings: `'s'` vs `"s"`